site/documentation/content/api/tenant/index.md

```diff
@@ -201,13 +201,25 @@ This allows for future *well-known* additions and also allows to add further inf
 
 The table below contains the properties which are used to configure a tenant's resource limits:
 
-| Name                     | Mandatory | JSON Type     | Default Value | Description |
-| :------------------------| :-------: | :------------ | :------------ | :---------- |
-| *max-connections*        | *no*      | *number*      | `-1`          | The maximum number of concurrent connections allowed from devices of this tenant. The default value `-1` indicates that no limit is set. |
-| *data-volume*            | *no*      | *object*      | `-`           | The maximum data volume allowed for the given tenant. Refer to  [Data Volume Configuration Format]({{< relref "#data-volume-configuration-format" >}}) for details.|
+| Name                | Mandatory | JSON Type     | Default Value | Description |
+| :------------------ | :-------: | :------------ | :------------ | :---------- |
+| *max-connections*   | *no*      | *number*      | `-1`          | The maximum number of concurrent connections allowed from devices of this tenant. The default value `-1` indicates that no limit is set. |
+| *max-ttl*           | *no*      | *number*      | `-1`          | The maximum time-to-live (in seconds) to use for events published by  devices of this tenant. Any default TTL value specified at either the tenant or device level will be limited to the max value specified here. If this property is set to a value other than `-1` and no default TTL is specified for a device, the max value will be used for events published by the device. A value of `-1` (the default) indicates that no limit is set. |
+| *data-volume*       | *no*      | *object*      | `-`           | The maximum data volume allowed for the given tenant. Refer to  [Data Volume Configuration Format]({{< relref "#data-volume-configuration-format" >}}) for details.|
 
 Protocol adapters SHOULD use the *max-connections* property to determine if a device's connection request should be accepted or rejected.
 
+Protocol adapters SHOULD use the *max-ttl* property to determine the *effective time-to-live* for events published by devices as follows:
+
+1. If a non-default *max-ttl* is set for the tenant, use that value as the *effective ttl*, otherwise set *effective ttl* to `-1`.
+1. If the event published by the device
+   1. contains a *ttl* property and *effective ttl* is not `-1` and the *ttl* value provided by the device is smaller than the
+      *effective ttl*, use the device provided *ttl* value as the new *effective ttl*.
+   1. does not contain a *ttl* property but a default *ttl* value is configured for the device (either at the tenant or the device level)
```

Comment:
Speaking of the default "ttl" on the tenant or device level: 
I think it would be helpful to have this `ttl` JSON object in the `defaults` object be explicitly documented in the Tenant Information Format and the Device Registry Docs, as well as in the device-registry-v1.yaml. 
Doesn't have to be done in this PR though.

Reply:
The default properties are a very generic construct and I am not sure, if we want to prescribe any semantics in the Tenant API. Also, the adapter user guides already define, which default properties they support ...

Reply:
Ah, yes, didn't check the adapter user guides. It's not very prominent there though.
And the restriction on the value not being adopted if max-ttl is set is not mentioned there.
Having it all together in the device-registry-v1.yaml with cross references between ttl and max-ttl sounds more user-friendly to me. But again, this can be done in a separate PR.